### PR TITLE
feat(nft-user-activity): Add user activity table and fetches

### PR DESCRIPTION
**Pull request by ChefHutch**
:bin:

### CODE REVIEW

File: .env.development [hunk 1]
```diff
@@ -16,8 +16,7 @@ REACT_APP_NODE_4 = "https://nodes.pancakeswap.com/"
 REACT_APP_GRAPH_API_PROFILE = "https://api.thegraph.com/subgraphs/name/pancakeswap/profile"
 REACT_APP_GRAPH_API_PREDICTION = "https://api.thegraph.com/subgraphs/name/pancakeswap/prediction-v2"
 REACT_APP_GRAPH_API_LOTTERY = "https://api.thegraph.com/subgraphs/name/pancakeswap/lottery"
-REACT_APP_GRAPH_API_NFT_MARKET = "https://api.thegraph.com/subgraphs/name/chefnyan/nft-markets"
-
+REACT_APP_GRAPH_API_NFT_MARKET = "https://api.thegraph.com/subgraphs/name/pancakeswap/nft-market"
```
Comment (ChefHutch) [inline .env.development:19]: New deploy on this url courtesy of Kai

File: src/state/nftMarket/hooks.ts [hunk 2]
```diff
@@ -40,57 +37,3 @@ export const useGetNFTInitializationState = () => {
...
-  }
-  return []
-}
-
-/**
- * Fetch metadata for a given array of Nft subgraph responses
- * @param sgNfts NftTokenSg[]
- * @returns NFT[]
- */
-export const useGetNftMetadata = (sgNfts: TokenMarketData[], account?: string) => {
```
Comment (ChefHutch) [inline src/state/nftMarket/hooks.ts:69]: This is unused

File: src/state/nftMarket/hooks.ts [hunk 3]
```diff
@@ -40,57 +37,3 @@ export const useGetNFTInitializationState = () => {
 export const useUserNfts = (): UserNftsState => {
   return useSelector((state: State) => state.nftMarket.data.user)
 }
-
-export const useUserActivity = (): (Transaction | AskOrder)[] => {
```
Comment (ChefHutch) [inline src/state/nftMarket/hooks.ts:44]: Migrated into its own hook and modified lots

File: src/views/Nft/market/Profile/components/ActivityHistory/ActivityRow.tsx [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    day: 'numeric',
+    hour: 'numeric',
+    minute: 'numeric',
+  })
+  const [onPresentMobileModal] = useModal(
+    <MobileModal nft={nft} activity={activity} localeTimestamp={localeTimestamp} bnbBusdPrice={bnbBusdPrice} />,
+  )
+
+  return (
+    <tr {...((isXs || isSm) && { onClick: onPresentMobileModal })}>
```
Comment (ChefHutch) [inline src/views/Nft/market/Profile/components/ActivityHistory/ActivityRow.tsx:43]: Modal is only to be opened on smaller device sizes

File: src/config/localization/translations.json [hunk 5]
```diff
@@ -1552,5 +1552,21 @@
...
+  "Buy and Sell verified PancakeSwap collectibles.": "Buy and Sell verified PancakeSwap collectibles.",
+  "Modified": "Modified",
+  "From/To": "From/To",
+  "Item": "Item",
+  "PancakeSwap NFTs only... for now!": "PancakeSwap NFTs only... for now!",
+  "Bought": "Bought",
+  "Event": "Event",
+  "Manage/Sell": "Manage/Sell",
+  "Sold": "Sold",
+  "Page %currentPage% of %maxPages%": "Page %currentPage% of %maxPages%"
```
Comment (Chef-Cheems) [inline src/config/localization/translations.json:1571]: There's already `Page %page% of %maxPage%`

File: src/config/localization/translations.json [hunk 6]
```diff
@@ -1552,5 +1552,21 @@
   "Removing it will suspend your profile, and you won’t be able to earn points, participate in team activities, or be eligible for new NFT drops.": "Removing it will suspend your profile, and you won’t be able to earn points, participate in team activities, or be eligible for new NFT drops.",
   "Traits": "Traits",
   "Trait": "Trait",
-  "Count": "Count"
+  "Count": "Count",
+  "$%amount%": "$%amount%",
```
Comment (Chef-Cheems) [inline src/config/localization/translations.json:1556]: lol 🤣 
Is it needed? I think we've been putting $ before number in 9999 other places all this time

**Review by Chef-Cheems — APPROVED** [review]

File: src/views/Nft/market/Profile/components/ActivityHistory/ActivityPrice.tsx [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+  return (
+    <Flex flexDirection="column" alignItems="flex-end">
+      {price ? (
+        <>
+          <Flex justifySelf="flex-start" alignItems="center">
+            <BinanceIcon width="12px" height="12px" mr="4px" />
+            <Text bold>{price}</Text>
+          </Flex>
+          <Text fontSize="12px" color="textSubtle">
+            {`(~${t('$%amount%', {
```
Comment (hachiojidev) [inline src/views/Nft/market/Profile/components/ActivityHistory/ActivityPrice.tsx:19]: Doesn't this result in two $$?

Comment (Chef-Cheems) [reply src/views/Nft/market/Profile/components/ActivityHistory/ActivityPrice.tsx:19]: I don't think so, first $ is the template string variable thingy `${x}`
But in bunch of other places we do have additional dollar sign here, e.g. CostLabel
```javascript
<Flex alignItems="center" {...props}>
      {priceInUsd > 0 && (
        <Text fontSize="12px" color="textSubtle">{`($${priceInUsd.toLocaleString(undefined, {
          minimumFractionDigits: 2,
          maximumFractionDigits: 2,
        })})`}</Text>
      )}
      <BNBAmountLabel amount={cost} />
    </Flex>
```

Comment (hachiojidev) [reply src/views/Nft/market/Profile/components/ActivityHistory/ActivityPrice.tsx:19]: Haha you're right.

File: src/config/localization/translations.json (hunk 6 above)
Comment (ChefHutch) [reply src/config/localization/translations.json:1556]: Listen it's either get told off by @Chef-Cheems for doing this - or get told off by @memoyil for not doing it! 

Agree it's overkill. I'm sure people who put the symbol after the number can live without it.

**Comment by ChefHutch** [comment]
https://github.com/pancakeswap/pancake-frontend/pull/2321

**Pull request closed without merging**